### DISCUSSION TITLE
custom select in lucid relationships

### DISCUSSION THREAD

**Discussion by farshadfahimi** [adonisjs/core#1398] [Help]
I have a modl like below and need to append the `slug` in the relation select
The `product` model has `slug`.
```javascript
class Product extends Model
{
       media() {
            return this.belongsToMany('App/Models/ProductsMedia', 'products_id',                  'media_id')
            .pivotTable('products_media_xref')
            .withPivot(['order', 'is_default', 'is_published'])
      }
}

class ProductsMedia extends Model {
   static get computed() {
    return ['text'];
  }

  getText() {
    console.log(this.slug)  // I have slug in the product model and it show undefined 
  }
}
```

in my `productsMedia` table I have `id,type` and I need to access the `product.slug` in the this model with select the `slug` in relation query. but it change my query even if I use the `raw query`.
I'm access the media all the time with this relation so
How should I achive this?

**Comment by McSneaky** [comment]
Hmm, `this.slug` is undefined because it doesn't exist on **this** model 😉 
If you want to load it in from some other model you have to extend that other model or pass it down as an parameter to the function

**Reply by farshadfahimi** [reply]
I know that ‍`this.slug` is undefined. but if you check that.I want to select the `slug` in relation query, in this way I can access to the `slug` in ProductMedia Model but when I use select with `raw` or with out it `builder` append the `products_media` as a table to the query
like below
```javascript
class Product extends Model
{
 media()
{
return this.belongsToMany('App/Models/ProductsMedia', 'products_id', 'media_id')
            .select('*', DB.raw('products.slug'))
            .pivotTable('products_media_xref')
            .withPivot(['order', 'is_default', 'is_published'])
            .orderBy('is_default', 'desc')
            .orderBy('order', 'asc')
}
}

```

with this relation I have query like below

```
select `products_media`.*, `products_media`.`products`.`slug`, `products_media_xref`.`media_id` as `pivot_media_id`, `products_media_xref`.`products_id` as `pivot_products_id`, `products_media_xref`.`order` as `pivot_order`, `products_media_xref`.`is_default` as `pivot_is_default`, `products_media_xref`.`is_published` as `pivot_is_published` from `products_media` inner join `products_media_xref` on `products_media`.`id` = `products_media_xref`.`media_id` where `products_media_xref`.`products_id` = 21436 order by `is_default` desc, `order` asc - ER_BAD_FIELD_ERROR: Unknown column 'products_media.products.slug' in 'field list' 
```

in the generated query with `raw` in the relation I think it should be like below
```
select `products_media`.*, `products`.`slug`, `products_media_xref`.`media_id` as `pivot_media_id`, 
```

but it append the `products_media` before the `products.slug`

**Reply by thetutlage** [reply]
Yes, `products_media` shouldn't be prepended. Mind doing a PR for the fix?

**Reply by farshadfahimi** [reply]
I like to, but I'm beginner and didn't have good knowledge in framework, so it's better to some one else do this.
maybe in feature I could help more with better knowledge

**Comment by farshadfahimi** [comment]
@McSneaky 

right know How should pass the data to model in the relation. because I'm not in the controller to create an instance and assign data to it
